tests/integration/acp.go

```diff
@@ -310,11 +310,192 @@ func addDocActorRelationshipACP(
 	}
 }
 
+// DeleteDocActorRelationship will attempt to delete a relationship between a document and an actor.
+type DeleteDocActorRelationship struct {
+	// NodeID may hold the ID (index) of the node we want to delete doc actor relationship on.
+	//
+	// If a value is not provided the relationship will be deleted on all nodes, unless testing with
+	// sourcehub ACP, in which case the relationship will only be deleted once.
+	NodeID immutable.Option[int]
+
+	// The collection in which the target document we want to delete relationship for exists.
+	//
+	// This is a required field. To test the invalid usage of not having this arg, use -1 index.
+	CollectionID int
+
+	// The index-identifier of the document within the collection.  This is based on
+	// the order in which it was created, not the ordering of the document within the
+	// database.
+	//
+	// This is a required field. To test the invalid usage of not having this arg, use -1 index.
+	DocID int
+
+	// The name of the relation within the relationship we want to delete (should be defined in the policy).
+	//
+	// This is a required field.
+	Relation string
+
+	// The target public identity, i.e. the identity of the actor with whom the relationship is with.
+	//
+	// This is a required field. To test the invalid usage of not having this arg, use -1 index.
+	TargetIdentity int
+
+	// The requestor identity, i.e. identity of the actor deleting the relationship.
+	// Note: This identity must either own or have managing access defined in the policy.
+	//
+	// This is a required field. To test the invalid usage of not having this arg, use -1 index.
+	RequestorIdentity int
+
+	// Result returns true if the relationship record was expected to be found and deleted,
+	// and returns false if no matching relationship record was found (no-op).
+	ExpectedRecordFound bool
+
+	// Any error expected from the action. Optional.
+	//
+	// String can be a partial, and the test will pass if an error is returned that
+	// contains this string.
+	ExpectedError string
+}
+
+func deleteDocActorRelationshipACP(
+	s *state,
+	action DeleteDocActorRelationship,
+) {
+	if action.NodeID.HasValue() {
```

Comment:
todo: This seems like an unmanagably large if-else block and it looks like there is a fair amount of duplication.  Please rework this function.

Reply:
This is an issue since the previous node index bug was fixed, I will be taking care of this in https://github.com/sourcenetwork/defradb/issues/3076

Won't do in this PR.

Reply:
If you really don't want to tidy it up a bit in this function can you please add a todo comment linked to that ticket and make sure this function is mentioned in the issue?

Reply:
> If you really don't want to tidy it up a bit in this function can you please add a todo comment linked to that ticket and make sure this function is mentioned in the issue?

I can if you really like, but I literally have the fix I am working on in a separate branch so will be opening the clean up PR soon

Reply:
Ah okay, if it is all coded up already somewhere nearly ready to go dont bother here :)

Reply:
Cheers